### DISCUSSION TITLE
Is it possible to mimic isort's "From First" behavior?

### DISCUSSION THREAD

**Discussion by codekoala** [astral-sh/ruff#6449] [Q&A]
Several of my projects have `isort` configured with `from_first=true` (https://pycqa.github.io/isort/docs/configuration/options.html#from-first). This results in something like this:

```python
import csv
from random import randint
from datetime import datetime
```

being formatted as

```python
from datetime import datetime
from random import randint
import csv
```

Most of the codebase follows this pattern because of our style guide and pre-commit hooks.

I haven't figured out how to configure `ruff` to maintain this style. The best result I have right now is:

```python
import csv
from datetime import datetime
from random import randint
```

**Comment by zanieb** [comment]
In general, we're hesitant to add configuration options. I feel like the vast majority of projects use the default order. It'd be nice for us to stick to enforcing a consistent ordering in the ecosystem if possible. Could you share why this order is better for you? (other than it's what you're doing already)

I'm happy to hear from other community members too!

**Comment by codekoala** [comment]
I suppose it's just for historical reasons, from before the time we had intelligent helpers to sort our imports--`from` comes before `import` alphabetically. Imports could be sorted using more simplistic, not Python-specific/aware options.
